internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -764,20 +764,22 @@ ExecExprVisitor::ExecTermVisitorImpl(TermExpr& expr_raw) -> BitsetType {
     }
 
     if (use_pk_index) {
-        auto id_array = std::make_unique<IdArray>();
+        std::shared_ptr<IdArray> id_array;
         switch (field_meta.get_data_type()) {
             case DataType::INT64: {
-                auto dst_ids = id_array->mutable_int_id();
+                auto builder = arrow::Int64Builder();
                 for (const auto& id : expr.terms_) {
-                    dst_ids->add_data((int64_t&)id);
+                    builder.Append(id);
                 }
+                id_array = builder.Finish().ValueOrDie();
                 break;
             }
             case DataType::VARCHAR: {
-                auto dst_ids = id_array->mutable_str_id();
+                auto builder = arrow::StringBuilder();
```

Comment:
This part of the code is redundant

there is a method  `ExecExprVisitor::ExecTermVisitorImpl<std::string>` below, so this part will never be used 😅